embedded-graphics/src/mock_display.rs

```diff
@@ -324,19 +403,32 @@ where
     type Error = core::convert::Infallible;
 
     fn draw_pixel(&mut self, pixel: Pixel<C>) -> Result<(), Self::Error> {
-        let Pixel(Point { x, y }, color) = pixel;
-        if !(0..SIZE).contains(&(x as usize)) || !(0..SIZE).contains(&(y as usize)) {
-            return Ok(());
+        let Pixel(point, color) = pixel;
+
+        if !DISPLAY_AREA.contains(point) {
+            if self.allow_out_of_bounds_drawing {
+                return Ok(());
+            } else {
+                panic!(
+                    "tried to draw pixel outside the display area (x: {}, y: {})",
+                    point.x, point.y
+                );
+            }
         }
 
-        let i = x + y * SIZE as i32;
-        self.0[i as usize] = Some(color);
+        if !self.allow_overdraw {
+            if self.get_pixel(point).is_some() {
+                panic!("tried to draw pixel twice (x: {}, y: {})", point.x, point.y);
+            }
+        }
+
+        self.set_pixel(point, Some(color));
```

Comment:
Clippy is complaining that this could be the following, but I understand if you want to keep it as is for consistency with the above `if { if { ... } }`

```suggestion
        if !self.allow_overdraw && self.get_pixel(point).is_some() {
            panic!("tried to draw pixel twice (x: {}, y: {})", point.x, point.y);
        }
```

Reply:
I've changed it to make clippy happy.